bin/network-monitor/src/counter.rs

```diff
@@ -81,43 +80,98 @@ async fn fetch_counter_value(
     account_id: AccountId,
 ) -> Result<Option<u64>> {
     let id_bytes: [u8; 15] = account_id.into();
-    let req = miden_node_proto::generated::account::AccountId { id: id_bytes.to_vec() };
-    let resp = rpc_client.get_account_details(req).await?.into_inner();
-    if let Some(raw) = resp.details {
-        let account = Account::read_from_bytes(&raw)
-            .map_err(|e| anyhow::anyhow!("failed to deserialize account details: {e}"))?;
+    let account_id_proto =
+        miden_node_proto::generated::account::AccountId { id: id_bytes.to_vec() };
+
+    // Request account details with storage header (but no storage maps needed)
+    let request = miden_node_proto::generated::rpc_store::AccountRequest {
+        account_id: Some(account_id_proto),
+        block_num: None,
+        details: Some(
+            miden_node_proto::generated::rpc_store::account_request::AccountDetailRequest {
+                code_commitment: None,
+                asset_vault_commitment: None,
+                storage_maps: vec![],
+            },
+        ),
+    };
 
-        let storage_slot = account.storage().slots().first().expect("storage slot is always value");
-        let word = storage_slot.value();
-        let value = word.as_elements().last().expect("a word is always 4 elements").as_int();
+    let resp = rpc_client.get_account(request).await?.into_inner();
 
-        Ok(Some(value))
-    } else {
-        Ok(None)
+    // Extract the counter value from the storage header
+    if let Some(details) = resp.details {
+        let storage_details =
+            details.storage_details.context("missing storage details")?;
+
+        let storage_header = storage_details.header.context("missing storage header")?;
+
+        // The counter value is in the first storage slot (index 0)
+        let first_slot = storage_header.slots.first().context("no storage slots found")?;
+
+        let commitment =
+            first_slot.commitment.as_ref().context("missing storage slot commitment")?;
+
+        // The commitment is a Word containing the counter value
+        // For a value slot, the Word directly contains the value
+        let word: Word = (*commitment).try_into().context("failed to convert commitment to word")?;
```

Comment:
nit: Conceptually this slot is a value slot (ie, not a storage map slot) so this is not a commitment but rather just a plain `Word` (I think the protobuf comments might also have this detail)